### DISCUSSION TITLE
How to move elements into one row sorted by length?

### DISCUSSION THREAD

**Discussion by Dip115** [cwapi3d/cwapi3dpython#209] [Q&A]
Hello!
I have a problem with moving elements around. I have a script that moves selected elements to one point.
But there's no way I can make it so that the elements are next to each other instead of overlapping. And I would like them to be sorted by length. Can anyone tell me how to solve this problem? Thanks!

**Comment by Brunner246** [comment]
@Dip115 

Please post an operational code snippet for specific questions.
you have to find the distance between the elements according to their width and then place them accordingly. 

Sorting example with python https://docs.python.org/3/howto/sorting.html
```python
import element_controller as ec
import attribute_controller as ac
import geometry_controller as gc

element_ids = ec.get_active_identifiable_element_ids()

sort_elements_by_length = sorted(element_ids, key= lambda element : gc.get_length(element))

for element in sort_elements_by_length :
    print(gc.get_length(element))

    
>>> 500.0
>>> 500.0
>>> 700.0
>>> 6789.999999999996
>>> 17586.00000089407

````

**Comment by ancelin-charpentier** [comment]
Hey @Dip115 ,
I had to do something similar. Best solution I found was to create a dictionary using the ref you want as a key (The length in your case but I guess you would have to change them to a INT)  
Then I created a list using the keys but sorted in the order I wanted (By this I mean you can reverse as I guess that is your intent)
Then I run a loop over my list checking if the value of reference (an INT of the length in your case) is matching the length I got from my active element. (Once again here you would have to compare an INT against an INT)
After this you can locate your element and iterate your Z value. 

I hope that will help,

**Reply by Dip115** [reply]
Hello! Thank you!

**Reply by ancelin-charpentier** [reply]
No problem, there is quite a bit do there but it is a good project! Let me know if you make it, otherwise and if you are interested, we can do a videocall so I can show you the code I have made and how it goes.
